lib/npg_qc/autoqc/checks/upstream_tags.pm

```diff
@@ -129,7 +129,7 @@ has 'tag0_bam_file'  => ( isa        => 'Str',
                         );
 sub _build_tag0_bam_file {
   my $self = shift;
-  my $basefilename = sprintf q[%s_%s#0.bam], $self->id_run, $self->position;
+  my $basefilename = sprintf q[%s_%s#0.bam],$self->get_id_run(),$self->composition->get_component(0)->position();
```

Comment:
get_run_id returns an undefined value if not all components have the same id_run value. Therefore, it might make sense to introduce an early check (in BUILD method?) and raise an error if not all components belong to the same run.